ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InputsAndEffects.scala

```diff
@@ -106,57 +102,50 @@ private[kvutils] object InputsAndEffects {
   }
 
   /** Compute the effects of a DAML transaction, that is, the created and consumed contracts. */
-  def computeEffects(entryId: DamlLogEntryId, tx: SubmittedTransaction): Effects = {
+  def computeEffects(entryId: DamlLogEntryId, tx: Transaction.AbsTransaction): Effects = {
     // TODO(JM): Skip transient contracts in createdContracts/updateContractKeys. E.g. rewrite this to
     // fold bottom up (with reversed roots!) and skip creates of archived contracts.
     tx.fold(Effects.empty) {
       case (effects, (nodeId, node)) =>
         node match {
-          case fetch: NodeFetch[ContractId] =>
+          case fetch @ NodeFetch(_, _, _, _, _, _) =>
             effects
-          case create: NodeCreate[ContractId, VersionedValue[ContractId]] =>
+          case create @ NodeCreate(_, _, _, _, _, _, _) =>
             effects.copy(
-              createdContracts =
-                relativeContractIdToStateKey(entryId, create.coid.asInstanceOf[RelativeContractId]) -> create
-                  :: effects.createdContracts,
+              createdContracts = contractIdToStateKey(create.coid) -> create :: effects.createdContracts,
               updatedContractKeys = create.key
                 .fold(effects.updatedContractKeys)(
                   keyWithMaintainers =>
-                    effects.updatedContractKeys +
+                    effects.updatedContractKeys.updated(
                       (globalKeyToStateKey(
                         GlobalKey(
                           create.coinst.template,
-                          forceNoContractIds(keyWithMaintainers.key.value))) ->
-                        DamlContractKeyState.newBuilder
-                          .setContractId(encodeRelativeContractId(
-                            entryId,
-                            create.coid.asInstanceOf[RelativeContractId]))
-                          .build))
+                          // FIXME: We probably should not crash here.
```

Comment:
preferably throw a kvutils `Err` here instead as the caller assumes only those are expected.

Reply:
done.